tests/components/modbus/conftest.py

```diff
@@ -201,3 +201,26 @@ async def base_config_test(
         config_modbus=config_modbus,
         expect_init_to_fail=expect_init_to_fail,
     )
+
+
+async def run_service_update(hass, config, entity_id, reuse=False):
+    """Run test for service write_coil."""
+
+    if not reuse:
+        config_modbus = {
+            DOMAIN: {
+                CONF_NAME: DEFAULT_HUB,
+                CONF_TYPE: "tcp",
+                CONF_HOST: "modbusTest",
+                CONF_PORT: 5001,
+                **config,
+            },
+        }
+        assert await async_setup_component(hass, DOMAIN, config_modbus)
+        await hass.async_block_till_done()
+        assert await async_setup_component(hass, "homeassistant", {})
+        await hass.async_block_till_done()
+    await hass.services.async_call(
```

Comment:
Please inline the service call in the tests instead of having it in this helper.

Reply:
Done, it is identical for all tests, which was the reason had it in the helper. Without the service_call the helper becomes a bit more simple, parameter "reuse" is gone.

Please do not forget this PR depends on #49917 (the actual production code change).

Reply:
I think we can merge this PR. It's small enough. We can close the other PR.